src/js/actions/libraries.js

```diff
@@ -297,12 +297,16 @@ define(function (require, exports) {
                     
                 return descriptor.playObject(createObj);
             })
+            // Wait for 1 second before Photoshop changes layer's smart object type from linked SO to 
+            // cloud-linked SO
+            // 
+            // FIXME: Instead of waiting, Photoshop should emit an event to tell DS that a layer's smart object 
```

Comment:
I created a watson bug https://watsonexp.corp.adobe.com/#bug=4073805 and included in the comment below.